go-fuzz/master.go

```diff
@@ -71,25 +71,58 @@ func masterLoop(m *Master) {
 			log.Printf("slave %v died", s.id)
 			delete(m.slaves, id)
 		}
+		m.mu.Unlock()
 
 		// Print stats line.
-		uptime := time.Since(m.startTime)
-		lastInput := time.Since(m.lastInput)
-		restarts := uint64(0)
-		if m.statExecs != 0 {
-			restarts = m.statExecs / m.statRestarts
-		}
-		procs := 0
-		for _, s := range m.slaves {
-			procs += s.procs
-		}
-		log.Printf("slaves: %v, corpus: %v (%v ago), crashers: %v,"+
-			" restarts: 1/%v, execs: %v (%.0f/sec), cover: %v, uptime: %v",
-			procs, len(m.corpus.m), fmtDuration(lastInput), len(m.crashers.m),
-			restarts, m.statExecs, float64(m.statExecs)*1e9/float64(uptime),
-			m.coverFullness, fmtDuration(uptime))
-		m.mu.Unlock()
+		stats := m.masterStats()
+		log.Println(stats.String())
+
+		broadcastStats(stats)
+	}
+}
+
+func (m *Master) masterStats() masterStats {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	stats := masterStats{
+		Corpus:           uint64(len(m.corpus.m)),
+		Crashers:         uint64(len(m.crashers.m)),
+		StartTime:        m.startTime,
+		LastMutationTime: m.lastInput,
```

Comment:
s/LastMutationTime/LastNewInputTime/
mutations happen all the time, this is the time when we generated new interesting input and added it to corpus